spec/emerald_spec.rb

```diff
@@ -0,0 +1,129 @@
+require 'spec_helper'
+require 'tempfile'
+require 'helper/user_comms'
+include Emerald
+require 'emerald/parser'
+require 'emerald/compiler'
+
```

Comment:
I think it depends - I like the given, when, then structure when writing tests, and I think with RSpec contexts provide a reasonable means of adhering to this (though I agree, they do clutter sometimes, depending on where they're used). Same with before and after hooks - it depends on the value that their contents have on individual test readability, but their use can be helpful to dry up your tests.